components/org.wso2.carbon.identity.conditional.auth.functions.entgra/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/entgra/GetDeviceInfoEntgraFunctionImpl.java

```diff
@@ -0,0 +1,256 @@
+                .build();
+        client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+
+    }
+
+    @Override
+    public void getDeviceInfoEntgra(JsAuthenticationContext context, String platformOS, String deviceID,
+                                    Map<String, Object> eventHandlers) throws EntgraConnectorException {
+
+        try {
+            JsAuthenticatedUser user = Util.getUser(context);
+            String tenantDomain = user.getWrapped().getTenantDomain();
+            String username = user.getWrapped().getUserName();
+
+            // Getting connector configurations.
+            String clientKey = CommonUtils.getConnectorConfig(Constants.CLIENT_KEY, tenantDomain);
+            String clientSecret = CommonUtils.getConnectorConfig(Constants.CLIENT_SECRET, tenantDomain);
+            String tokenURL = CommonUtils.getConnectorConfig(Constants.TOKEN_URL, tenantDomain);
+            String deviceInfoBaseURL = CommonUtils.getConnectorConfig(Constants.DEVICE_INFO_URL, tenantDomain);
+
+            String deviceInfoURL = deviceInfoBaseURL + "/" + platformOS + "/" + deviceID;
+
+            AsyncProcess asyncProcess = new AsyncProcess((authenticationContext, asyncReturn) -> {
+                String outcome;
+                JSONObject response = null;
+
+                try {
+                    HttpPost tokenRequest = getTokenRequest(tokenURL, clientKey, clientSecret);
+
+                    // For catching and logging errors.
+                    String errorURL = tokenURL;
+                    try (CloseableHttpResponse tResponse = client.execute(tokenRequest)) {
+                        int tokenResponseCode = tResponse.getStatusLine().getStatusCode();
+
+                        if (tokenResponseCode >= 200 && tokenResponseCode < 300) {
+                            String tJsonString = EntityUtils.toString(tResponse.getEntity());
+                            JSONParser parser = new JSONParser();
+                            JSONObject jsonTokenResponse = (JSONObject) parser.parse(tJsonString);
+                            String accessToken = (String) jsonTokenResponse.get(Constants.ACCESS_TOKEN);
+
+                            HttpGet deviceInfoRequest = getDeviceInfoRequest(deviceInfoURL, accessToken);
+
+                            tResponse.close();
+                            try (CloseableHttpResponse dResponse = client.execute(deviceInfoRequest)) {
+                                int dResponseCode = dResponse.getStatusLine().getStatusCode();
+
+                                if (dResponseCode >= 200 && dResponseCode < 300) {
+                                    String dJsonString = EntityUtils.toString(dResponse.getEntity());
+                                    JSONObject jsonDeviceInfoResponse = (JSONObject) parser.parse(dJsonString);
+                                    String enrolledUser = (String) ((JSONObject) jsonDeviceInfoResponse.get("enrolmentInfo")).get("owner");
```

Comment:
Many of the lines in this class exceed 120 characters. Please limit them to 120,

Reply:
Done.